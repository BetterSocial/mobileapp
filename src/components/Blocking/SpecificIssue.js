import * as React from 'react';
import IconFA5 from 'react-native-vector-icons/FontAwesome5';
import RBSheet from 'react-native-raw-bottom-sheet';
import {ActivityIndicator, StyleSheet, Text, TextInput, TouchableOpacity, View} from 'react-native';

import {Button} from '../Button';
import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const SpecificIssue = ({refSpecificIssue, onPress, onSkip, loading}) => {
  const [message, setMessage] = React.useState('');
  return (
    <RBSheet
      ref={refSpecificIssue}
      closeOnDragDown={true}
      closeOnPressMask={true}
      customStyles={{
        container: styles.container,
        draggableIcon: styles.draggableIcon
      }}>
      <View>
        <Text style={styles.title}>Please specify the issue</Text>
        <TextInput
          style={styles.input}
          value={message}
          onChangeText={(v) => setMessage(v)}
          textAlignVertical="top"
          multiline
          placeholder={'Please provide more details to inform our\n team (min. 50 characters)'}
          placeholderTextColor={COLORS.gray400}
        />
        <TouchableOpacity testID="button-skip-test" style={styles.btnSkip} onPress={() => onSkip()}>
          <Text style={styles.btnSkipText}>Skip & just block this account</Text>
          <IconFA5 name="chevron-right" size={17} color={COLORS.black} />
        </TouchableOpacity>
        <View style={styles.containerBtn}>
          <Button testID="button-report-test" onPress={() => onPress(message)}>
            {loading ? (
              <ActivityIndicator testID="loading-indicator-test" />
            ) : (
              <Text style={styles.containerBtnText}>File Report</Text>
            )}
          </Button>
        </View>
      </View>
    </RBSheet>
  );
};

export default SpecificIssue;

const styles = StyleSheet.create({
  title: {
    fontFamily: fonts.inter[700],
    fontSize: 18,
    color: COLORS.black,
    marginLeft: 21,
    marginTop: 18
  },
  containerBtn: {
    marginRight: 22,
    marginLeft: 18,
    marginBottom: 19
  },
  containerBtnText: {
    marginRight: 22,
    marginLeft: 18,
    marginBottom: 19,
    fontFamily: fonts.inter[500],
    fontSize: 16,
    color: COLORS.white
  },
  btn: {
    paddingLeft: 18,
    paddingRight: 22,
    paddingTop: 8
  },
  btnSkip: {
    backgroundColor: COLORS.gray100,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 17,
    paddingVertical: 18,
    marginVertical: 22
  },
  btnSkipText: {
    fontFamily: fonts.inter[700],
    fontSize: 14,
    color: COLORS.black
  },
  input: {
    backgroundColor: COLORS.gray100,
    marginLeft: 17,
    marginRight: 23,
    borderRadius: 8,
    height: 233,
    marginTop: 14,
    paddingVertical: 19,
    paddingRight: 13,
    paddingLeft: 19,
    fontFamily: fonts.inter[400],
    color: COLORS.white,
    borderWidth: 1,
    borderColor: COLORS.gray200
  },
  container: {
    height: 'auto',
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
    backgroundColor: COLORS.almostBlack
  },
  draggableIcon: {
    backgroundColor: COLORS.gray100,
    width: 60
  }
});
