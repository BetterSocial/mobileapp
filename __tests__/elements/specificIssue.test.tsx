import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import SpecificIssue from '../../src/elements/Blocking/SpecificIssue';

jest.mock('react-native', () => {
  return {
    StyleSheet: {
      create: jest.fn((e) => e),
    },
  };
});

configure({adapter: new Adapter()});
jest.useFakeTimers();

describe('component SpecificIssue', () => {
  it('SpecificIssue snapshot', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const component = shallow(
      <SpecificIssue onPress={() => {}} refSpecificIssue={myRef} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('SpecificIssue renders correctly', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const tree = renderer
      .create(<SpecificIssue onPress={() => {}} refSpecificIssue={myRef} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
