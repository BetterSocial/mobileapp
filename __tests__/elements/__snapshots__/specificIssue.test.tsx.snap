// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SpecificIssue SpecificIssue snapshot 1`] = `
<RBSheet
  animationType="none"
  closeDuration={200}
  closeOnDragDown={true}
  closeOnPressBack={true}
  closeOnPressMask={true}
  customStyles={
    Object {
      "container": Object {
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": "auto",
      },
      "draggableIcon": Object {
        "backgroundColor": "#E0E0E0",
        "width": 60,
      },
    }
  }
  dragFromTopOnly={false}
  height={260}
  keyboardAvoidingViewEnabled={false}
  minClosingHeight={0}
  onClose={null}
  onOpen={null}
  openDuration={300}
>
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Inter-Bold",
          "fontSize": 18,
          "marginLeft": 21,
          "marginTop": 18,
        }
      }
    >
      Please specify the issue
    </Text>
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Please provide more details to inform our
 team (min. 50 characters)"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#F5F5F5",
          "borderRadius": 8,
          "color": "#828282",
          "fontFamily": "Inter-Regular",
          "height": 233,
          "marginLeft": 17,
          "marginRight": 23,
          "marginTop": 14,
          "paddingLeft": 19,
          "paddingRight": 13,
          "paddingVertical": 19,
        }
      }
      textAlignVertical="top"
      underlineColorAndroid="transparent"
      value=""
    />
    <TouchableOpacity
      activeOpacity={0.2}
      delayLongPress={600}
      extraButtonProps={
        Object {
          "rippleColor": "transparent",
        }
      }
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 22,
          "paddingHorizontal": 17,
          "paddingVertical": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "Inter-Bold",
            "fontSize": 14,
          }
        }
      >
        Skip & just block this account
      </Text>
      <Icon
        color="#000"
        name="chevron-right"
        size={17}
      />
    </TouchableOpacity>
    <View
      style={
        Object {
          "marginBottom": 19,
          "marginLeft": 18,
          "marginRight": 22,
        }
      }
    >
      <Btn
        onPress={[Function]}
      >
        <Text>
          File Report
        </Text>
      </Btn>
    </View>
  </View>
</RBSheet>
`;
